fix(matrix): check invalid params and empty matrix in matrix update

laurent-laporte-pro opened the pull request.

laurent-laporte-pro committed:
test(matrix): correct and improve unit test on matrices

laurent-laporte-pro committed:
refactor(command): improve code quality in `CreateLinks`

laurent-laporte-pro committed:
fix(matrix): check invalid params and empty matrix in matrix update

ghost commented:
### Tests

*  **matrix:** correct and improve unit test on matrices ([0af185a](https://github.com/AntaresSimulatorTeam/AntaREST/commit/0af185a2d45983cc6c9d1cdceb2ac969687da18a))
*  **model:** correct unit tests with empty matrices ([2ca9e4c](https://github.com/AntaresSimulatorTeam/AntaREST/commit/2ca9e4ceea8fe522fc5f451db2e5a2ceaba7e9d6))


### Code Refactoring

*  **command:** improve code quality in `CreateLinks` ([cd3e777](https://github.com/AntaresSimulatorTeam/AntaREST/commit/cd3e777faaad790e0f0e854a48966dfc7a78bc09))


### Bug Fixes

*  **matrix:** check invalid params and empty matrix in matrix update ([e90adb3](https://github.com/AntaresSimulatorTeam/AntaREST/commit/e90adb357e6762df83da8d857a43217517254f85))
*  **model:** generate empty matrix if CSV file not found or empty ([1476b9d](https://github.com/AntaresSimulatorTeam/AntaREST/commit/1476b9d427d6fcc9d36af25e9b910622723a6414))
*  **matrix:** use the format "%.18g" which is more appropriate for integer or regular floats ([fa2dfc4](https://github.com/AntaresSimulatorTeam/AntaREST/commit/fa2dfc44eafb7b30dca2173e8779da088556293d))
*  **model:** it is now possible to edit a matrix when it is missing ([011014d](https://github.com/AntaresSimulatorTeam/AntaREST/commit/011014d987a9e1e950f37e8f8ada9a8b655fff4b))
*  **launcher:** correct the way the launcher is started (use a shell) ([5ee0b4a](https://github.com/AntaresSimulatorTeam/AntaREST/commit/5ee0b4ad8e8f88a16e0713e516dde4978c05c05e))
*  **model:** avoid using asynchronous matrix de-normalization in `execute_or_add_commands` ([4e38f7c](https://github.com/AntaresSimulatorTeam/AntaREST/commit/4e38f7c3e2309252efaf1dce0d8d0fdcb9a9b9f2))


### Documentation

*  **model:** improve the documentation of the `normalize` and `denormalize` methods ([ffdd6fe](https://github.com/AntaresSimulatorTeam/AntaREST/commit/ffdd6fedcd834fb74e14cc8e53bacd337d6316c6))


### Contributors

<a href="https://github.com/laurent-laporte-pro">laurent-laporte-pro</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

laurent-laporte-pro committed:
fix(model): generate empty matrix if CSV file not found or empty

To allow user modification, generate an empty matrix if the CSV file is not found or empty. This empty matrix can be used as a placeholder for future updates. The commit also handles cases where `FileNotFoundError` or `EmptyDataError` occur, ensuring that an empty matrix is generated in those situations.

laurent-laporte-pro committed:
fix(matrix): use the format "%.18g" which is more appropriate for integer or regular floats

laurent-laporte-pro committed:
fix(model): it is now possible to edit a matrix when it is missing

laurent-laporte-pro committed:
test(model): correct unit tests with empty matrices

laurent-laporte-pro committed:
fix(launcher): correct the way the launcher is started (use a shell)

laurent-laporte-pro committed:
docs(model): improve the documentation of the `normalize` and `denormalize` methods

laurent-laporte-pro committed:
fix(model): avoid using asynchronous matrix de-normalization in `execute_or_add_commands`

In a previous version, de-normalization was performed asynchronously. However, this cause problems with concurrent file access, especially when de-normalizing a matrix (which can take time).
To avoid this concurrency problem, it would be necessary to implement a locking system for the entire study using a file lock (since multiple processes, not only multiple threads, could access the same content simultaneously).
Currently, we use a synchronous call to address the concurrency problem within the current process (not across multiple processes)...
